### PR TITLE
Improve all edge cases around double timer issues

### PR DESCRIPTION
**Pull request by Raynos**
I think this also fixes the stale tombstones issue we see
in production.

r @kriskowal @jcorbin @shannili

### CODE REVIEW

File: node/out_request.js [hunk 1]
```diff
@@ -486,13 +500,27 @@ TChannelOutRequest.prototype.hookupCallback = function hookupCallback(callback)
 TChannelOutRequest.prototype.onTimeout = function onTimeout(now) {
     var self = this;
 
+    if (self.err) {
+        self.channel.logger.warn('weird onTimeout for errored out request', {
```
Comment (kriskowal) [inline node/out_request.js]: s/weird/unexpected

File: node/out_response.js [hunk 2]
```diff
@@ -217,9 +217,10 @@ TChannelOutResponse.prototype.emitFinish = function emitFinish() {
             codeString: self.codeString,
             errorMessage: self.message,
             remoteAddr: self.inreq.connection.socketRemoteAddr,
-            state: self.state
-        });
+            state: self.state,
 
+            isOk: self.ok
```
Comment (kriskowal) [inline node/out_response.js]: In one sentence, clarify when `isOk` should be used instead of merely `ok`.

**Comment by kriskowal** [comment]
:ok:

File: node/out_response.js (hunk 2 above)
Comment (Raynos) [inline node/out_response.js]: No reason. I wrote hasResponse and isOk at the same time

File: node/argstream.js [hunk 3]
```diff
@@ -153,10 +153,12 @@ function OutArgStream() {
         self._handleFrameChunk(2, null);
     });
     self.arg3.on('finish', function onArg3Finish() {
-        self._handleFrameChunk(3, null);
-        self._flushParts(true);
-        self.finished = true;
-        self.finishEvent.emit(self);
+        if (!self.finished) {
```
Comment (ShanniLi) [inline node/argstream.js]: this smells bad. :(
arg3 shouldn't be finished twice at the first place ...

File: node/out_request.js (hunk 1 above)
Comment (ShanniLi) [inline node/out_request.js]: +1

File: node/out_response.js (hunk 2 above)
Comment (ShanniLi) [inline node/out_response.js]: yes, "ok" is better. Needs to keep the consistency.

**Comment by ShanniLi** [comment]
:+1:

**Pull request merged**